### PR TITLE
Octaneify

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Resolves: #79

This'd be a breaking change, but I see we're still below 1.0.0 :)
#Yolo :)

### CODE REVIEW

File: addon/components/copy-button.js [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
-  disabled: false,
+  disabled = false;
 
   /**
    * If true - scope event listener to this element
    * If false - scope event listener to document.body (ClipboardJS)
    * @property {Boolean} delegateClickEvent
    */
-  delegateClickEvent: true,
+  delegateClickEvent = true;
```
Comment (NullVoxPopuli) [inline addon/components/copy-button.js]: can we just pass the element instance to clipboardJS? would that be easier?

**Comment by jkusa** [comment]
@NullVoxPopuli thanks for getting started on this :) 

As you mentioned, we are pre-1.0.0. However, I feel we should signal the breaking changes. Perhaps a 1.0.0 release that drops some button attribute pass-throughs and older Ember support. Also, it might be nice to break this up into two releases, one for ember-auto-import and one for octane. Thoughts? (btw, is the support for `...attributes` ember 3.11 and up?)

Do you have cycles to continue on this? I have a local commit on top of your changes that resolves the test failures. Let me know your thoughts on proceeding.

**Comment by NullVoxPopuli** [comment]
>  Thoughts?
> Do you have cycles to continue on this?

I don't know if I have the time to do that :(
I could help finish up this PR, and add polyfills and such, but idk if I could do much more than that

> ...attributes ember 3.11 and up?

...attributes support starts in 3.4, but there are polyfills to earlier, if needed.

**Comment by jkusa** [comment]
> ...attributes support starts in 3.4

Oh right! I was conflating this with the splattributes + element modifiers update in 3.11.

In that case, 1.0.0 could be  >= 3.4LTS, ember-auto-import, and contract change.

My commit that passes ember-try on release, beta, & canary can be found here: https://github.com/jkusa/ember-cli-clipboard/commit/5a26db1f5250825614d9eb80a3dcfee219b7bb6f

Thanks again for your time on this.

File: addon/components/copy-button.js (hunk 1 above)
Comment (jkusa) [reply addon/components/copy-button.js]: The behavior of ClipboardJS is different when you pass a selector vs an element.

With a selector you get event delegation where the click event is handled at `document.body`. With an element reference, the click handler is attached to the element.

See https://github.com/jkusa/ember-cli-clipboard/pull/35 for reference

Comment (NullVoxPopuli) [reply addon/components/copy-button.js]: legit! is that situation in #35 described in documentation somewhere?

**Pull request closed without merging**